docs/viplab3.0/computation_result.md

```diff
@@ -1,141 +1,235 @@
+| identifier    | string (uuid) |must   | unique id for this artifact | |
+| type          |  string   | must      | the type of this artifact | The type may be used by the client to determine how to display the artifact| 
+
+
+#### file artifact JSON object
+|Key            |Value Type |Opt / Must |Description |Comment |
+|---------------|-----------|-----------|------------|--------|
+| path          | string    | must      | the path in which the content had been created during the computation | |
+| MIMEtype      | string    | must      | MIMEtype of the conten | | 
+| content       | string    | must      | base64url encoded content of the file | | 
+
+In Viplab 2.0, the following MIMEtypes where allowed. They contain some custom defined types, which need to be keep supported:
+
+* text/plain
+* text/html
+* text/uri-list": links to websides to be presented in SC
+* image/png"
+* application/x-vgf: generated graphics output in 'ViP graphics format'
+* application/x-vgf3: generated graphics output in 'ViP graphics format 3D'
+* application/x-vgfc: generated graphics output in 'ViP graphics format contour plot'.|
+
+
+#### s3file artifact JSON object
+|Key            |Value Type |Opt / Must |Description |Comment |
+|---------------|-----------|-----------|------------|--------|
+| path          | string    | must      | the path in which the content had been created during the computation | |
+| MIMEtype      | string    | must      | MIMEtype of the conten | | 
+| url           | string    | must      | url on which the content of the file can be retrieved | | 
+| size          | int       | must      | size in bytes of the file stored on the external system | |
+
+
+#### notifications artifact JSON object
+Each notifications artifact contains at least one summary and unlimited number of notification elemets providing more details
+
+|Key            |Value Type |Opt / Must |Description |Comment | AS |
+|---------------|-----------|-----------|------------|--------|----|
+| summary | string | must | summary of all info_element's: e.g. one warning, one error may have a summary "An error occured." | E.g.: "[Error] Backend has detected an error: no result!" or "Success!" |
+| elements |  array of notification objects | opt | the more detailed notification objects for this notifications object| |
+
+
+#### notification artifact JSON object
+|Key            |Value Type |Opt / Must |Description |Comment | AS |
+|---------------|-----------|-----------|------------|--------|----|
+| severity | string | must | One of {"error", "warning", "info"} | | |
+| type | string | must | One of {"system", "chain", "output", "callcheck", "interpreter", "compiler", "linker", "executable"} | For all chains: {"system, "chain", "output"}; for Octave/Matlab: {"callcheck", "interpreter"}; for C: {"callcheck", "compiler", "linker", "executable"}; for DuMuX: {"executable"} | If type equals "system", *whole* Result if of interest for a bug report. | Java?? |
```

Comment:
is this up to date with ViPLab 3.0?

Reply:
I would say it does fit. We did not port the backends over yet, but I don't see any issue in having those limitations.